### PR TITLE
chore: add api test

### PR DESCRIPTION
**Pull request by catngh**
**What does this PR do?**

- [ ] Setup fixture test for API
- [x] $nft query <>
- [ ] $ticker <>
- [ ] $balance
- [ ] tất cả các API (get list) config về role (nftrole, levelrole)
- [ ] $nft add
- [x] $nft recent

### CODE REVIEW

File: pkg/util/testhelper/db.go [hunk 1]
```diff
@@ -0,0 +1,84 @@
...
+		if err = db.Exec("REFRESH MATERIALIZED VIEW CONCURRENTLY view_nft_collection_stats").Error; err != nil {
+			l.Fatalf(err, "gorm: refresh materialized view 'view_nft_collection_stats' error")
+		}
+		// if err = db.Exec("REFRESH MATERIALIZED VIEW CONCURRENTLY view_nft_tokens").Error; err != nil {
+		// 	l.Fatalf(err, "gorm: refresh materialized view 'view_nft_tokens' error")
+		// }
+		if err = db.Exec("REFRESH MATERIALIZED VIEW CONCURRENTLY view_nft_collections_attributes").Error; err != nil {
+			l.Fatalf(err, "gorm: refresh materialized view 'view_nft_collections_attributes' error")
+		}
+	})
```
Comment (vdhieu) [inline pkg/util/testhelper/db.go:81]: Remove this if mochi doesn't have materialized view

File: pkg/handler/nfts_test.go [hunk 2]
```diff
@@ -0,0 +1,228 @@
...
+			},
+			Rarity: &response.IndexerNFTTokenRarity{
+				Rank:  0,
+				Score: "",
+				Total: 6771,
+			},
+			MetadataID: "",
+		},
+	}
+	indexerMock.EXPECT().GetNFTDetail("0xb54FF1EBc9950fce19Ee9E055A382B1219f862f0", "1").Return(&indexerRespPH, nil)
```
Comment (vdhieu) [inline pkg/handler/nfts_test.go:201]: you should move this into the each test case block.

```suggestion
	indexerMock.EXPECT().GetNFTDetail(tt.querySymbol, tt.queryTokenId).Return(&indexerRespPH, nil)
```

File: pkg/util/testhelper/db.go [hunk 3]
```diff
@@ -0,0 +1,68 @@
...
+			fmt.Println(err)
+			l.Fatalf(err, "failed to open database connection")
+			return
+		}
+
+		path, err := os.Getwd()
+		if err != nil {
+			l.Error(err, "unable to get dir")
+		}
+		fmt.Println(path) // for example /home/user
```
Comment (vdhieu) [inline pkg/util/testhelper/db.go:41]: remove logs

File: pkg/handler/nfts_test.go [hunk 4]
```diff
@@ -0,0 +1,228 @@
...
+		log      logger.Logger
+	}
+	tests := []struct {
+		name             string
+		fields           fields
+		wantCode         int
+		wantErr          error
+		wantResponsePath string
+	}{
+		// TODO: Add test cases.
```
Comment (vdhieu) [inline pkg/handler/nfts_test.go:40]: remove comments

File: pkg/handler/nfts_test.go [hunk 5]
```diff
@@ -0,0 +1,236 @@
...
+
+func TestHandler_GetNewListedNFTCollection(t *testing.T) {
+	cfg := config.LoadTestConfig()
+	db := testhelper.LoadTestDB("../../migrations/test_seed")
+	repo := pg.NewRepo(db)
+	log := logger.NewLogrusLogger()
+	s := pg.NewPostgresStore(&cfg)
+	entityMock := entities.New(cfg, log, repo, s, nil, nil, nil, nil, nil, nil, nil)
+
+	type fields struct {
```
Comment (nguyennh4522) [inline pkg/handler/nfts_test.go:29]: why don't use direct entities and log?

File: pkg/handler/nfts_test.go [hunk 6]
```diff
@@ -0,0 +1,236 @@
...
+			queryTokenId:    "1",
+			queryGuildId:    "",
+			expectedAddress: "0xb54FF1EBc9950fce19Ee9E055A382B1219f862f0",
+			wantIndexerResp: &response.IndexerGetNFTTokenDetailResponse{
+				Data: response.IndexerNFTTokenDetailData{
+					TokenID:           "1",
+					CollectionAddress: "0xb54FF1EBc9950fce19Ee9E055A382B1219f862f0",
+					Name:              "Portalhead #1",
+					Description:       "A Modern People Project.",
+					Amount:            "",
```
Comment (nguyennh4522) [inline pkg/handler/nfts_test.go:116]: should remove if it is zero value (similar with others)

File: .github/workflows/ci-test.yml [hunk 7]
```diff
@@ -11,7 +11,9 @@ jobs:
           go-version: "^1.18"
 
       - name: Init Test DB
-        run: make init
+        run: |
+          make init
+          make migrate-test
```
Comment (nguyennh4522) [inline .github/workflows/ci-test.yml:16]: `make init` contain `make migrate-test`, don't change it

**Review by vdhieu — APPROVED** [review]
LGTM

File: pkg/handler/nfts_test.go (hunk 5 above)
Comment (catngh) [reply pkg/handler/nfts_test.go:29]: we need a new `entity` to load in test config, repo, and service

File: pkg/handler/nfts_test.go (hunk 6 above)
Comment (catngh) [reply pkg/handler/nfts_test.go:116]: i can remove empty test inputs, but i think we should keep api response exact format even if they are empty

**Review by nguyennh4522 — APPROVED** [review]

**Review by trkhoi — APPROVED** [review]

**Pull request merged**